dev-util/pycharm-professional/pycharm-professional-2024.3.ebuild

```diff
@@ -0,0 +1,168 @@
+	)
+	!bundled-jdk? (
+		>=virtual/jre-1.8
+	)
+"
+
+if [[ "${PN}" == *-professional ]]; then
+RDEPEND+="
+	bundled-xvfb? (
+		x11-base/xorg-server[xvfb]
+	)
+"
+fi
+
+RESTRICT="test"
+
+QA_PREBUILT="opt/${PN}/*"
+
+src_prepare() {
+	default
+
+	rm -v  "${S}"/help/ReferenceCardForMac.pdf || die
+
+	rm -v "${S}"/plugins/python-ce/helpers/pydev/_pydevd_{bundle,frame_eval}/*{darwin,win32}* || die
+
+	if ! use amd64; then
+		rm -v  "${S}"/plugins/python-ce/helpers/pydev/pydevd_attach_to_process/attach_linux_amd64.so || die
+		if [[ -d "${S}"/lib/async-profiler/ ]]; then
+			rm -v  "${S}"/lib/async-profiler/amd64/libasyncProfiler.so || die
+		fi
+	fi
+	if ! use arm64; then
+		rm -v  "${S}"/plugins/python-ce/helpers/pydev/pydevd_attach_to_process/attach_linux_aarch64.so || die
+		if [[ -d "${S}"/lib/async-profiler/ ]]; then
+			rm -v  "${S}"/lib/async-profiler/aarch64/libasyncProfiler.so || die
+		fi
+	fi
+	if ! use x86; then
+		rm -v  "${S}"/plugins/python-ce/helpers/pydev/pydevd_attach_to_process/attach_linux_x86.so || die
+	fi
+
+	sed -i \
+		-e "\$a\\\\" \
+		-e "\$a#-----------------------------------------------------------------------" \
+		-e "\$a# Disable automatic updates as these are handled through Gentoo's" \
+		-e "\$a# package manager. See bug #704494" \
+		-e "\$a#-----------------------------------------------------------------------" \
+		-e "\$aide.no.platform.update=Gentoo" bin/idea.properties
+
+	if use bundled-jdk; then
+		for file in "jbr/bin/"*; do
+			patchelf --set-rpath '$ORIGIN/../lib' "${file}" || die
+		done
+		for file in "jbr/lib/"{libjcef.so,jcef_helper}; do
+			patchelf --set-rpath '$ORIGIN' "${file}" || die
+		done
+	else
+		rm -r "jbr" || die
+	fi
+
+	if [[ "${PN}" == *-professional ]]; then
+		if use bundled-xvfb; then
+			patchelf --set-rpath '$ORIGIN/../lib' "${S}"/plugins/remote-dev-server/selfcontained/bin/{Xvfb,xkbcomp}
+		else
+			rm -vr "${S}"/plugins/remote-dev-server/selfcontained/{bin/*,lib/,X11/xkb/} || die
+		fi
+	fi
+}
+
+src_configure() {
+	:;
+}
+
+src_compile() {
+	:;
+}
+
+src_install() {
+	local DIR="/opt/${PN}"
+	local JRE_DIR="jbr"
+
+	insinto "${DIR}"
+	doins -r ./*
+
+	fperms 755 "${DIR}"/bin/{format.sh,fsnotifier,inspect.sh,jetbrains_client.sh,ltedit.sh,pycharm,pycharm.sh,restarter}
+
+	if use bundled-jdk; then
+		fperms 755 "${DIR}/${JRE_DIR}"/bin/{java,javac,javadoc,jcmd,jdb,jfr,jhsdb,jinfo,jmap,jps,jrunscript,jstack,jstat,jwebserver,keytool,rmiregistry,serialver}
+		fperms 755 "${DIR}"/"${JRE_DIR}"/lib/{cef_server,chrome-sandbox,jcef_helper,jexec,jspawnhelper}
+	fi
+
+	if [[ "${PN}" == *-professional ]]; then
+		if use bundled-xvfb; then
+			fperms 755 "${DIR}"/plugins/remote-dev-server/selfcontained/bin/{Xvfb,xkbcomp}
+		else
+			dosym -r /usr/bin/Xvfb "${DIR}"/plugins/remote-dev-server/selfcontained/bin/Xvfb
+			dosym -r /usr/bin/xkbcomp "${DIR}"/plugins/remote-dev-server/selfcontained/bin/xcbcomp
+			dosym -r /usr/share/X11/xkb "${DIR}"/plugins/remote-dev-server/selfcontained/X11/xkb
```

Comment:
So this will conflict with xkeyboard-config package that gets pulled by xorg-server and can't be avoided. Therefore `+bundled-xvfb` will always fail to install due to file collisions. 

I imagine *Xvfb* is avoidable, but the ebuild should then have a blocker against `xorg-server[xvfb]`. I'd personally just depend on that instead and forget trying to bundle X with the package. This just... feels weird and unnecessarily complex?

Reply:
The first name is the target. So it would've linked the installed Xvfb into the location of the bundled Xvfb.